catalog/plugin/catalog-plugin-async/src/main/java/org/codice/ddf/catalog/plugin/async/api/internal/ProcessResource.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.plugin.async.api.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ *
+ * A {@code ProcessResource} represents a local or remote resource that will be used during processing
+ * by the {@link ProcessingFramework}.
+ */
+public interface ProcessResource {
```

Comment:
We needed this to be able to be easily translatable back to a `ContentItem`, so it resembles it in a lot of ways. The reason we didn't use `ContentItem` is because a `ProcessResource` may be remote, while a `ContentItem` should always be local. Made some modifications to this interface, but I think we are going to leave it largely as is in the interest of time and since it is experimental.